### PR TITLE
feat(testing): add Jest for unit testing /src

### LINKED ISSUE #42: Add Jest for unit testing
What are everyone's thoughts around adding Jest for unit test `src`? (Not the Next.js site, but that could come later)

Would you accept a PR if I added this?

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR adds Jest + [`ts-jest`](https://github.com/kulshekhar/ts-jest) to get basic unit testing support for `/src`.

image

Fixes #42

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: 'galex',
-  ignorePatterns: ['example-nextjs'],
+  ignorePatterns: ['example-nextjs', '__tests__'],
```
Comment (jsjoeio) [inline .eslintrc.js:3]: 🙏🏼 Please the `lint` step and ESLint

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ Scan through our [existing issues](https://github.com/slipHQ/run-wasm/issues) to
 #### Prerequisites
 
 Make sure you have the following installed in your development environment:
+
```
Comment (jsjoeio) [inline CONTRIBUTING.md:28]: 💄 I think this is leftover from my other PR, which was done through GitHub UI so it didn't get formatted correctly until now.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (jsjoeio) [inline README.md:1]: 💄 I think this is leftover from my other PR, which was done through GitHub UI so it didn't get formatted correctly until now.

File: __tests__/wasmClient.test.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { createRunWasmClient } from '../src/index'
+import { RunWasmClient } from '../src/RunWasmClient'
+
+describe('createRunWasmClient', () => {
+  it('should return an instance of WasmClient', () => {
+    const wasmClient = createRunWasmClient('python')
+
+    expect(wasmClient).toBeInstanceOf(RunWasmClient)
+  })
+})
```
Comment (jsjoeio) [inline __tests__/wasmClient.test.ts:10]: 👍🏼 Very very basic test to make the function returns what it says it's going to return

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+}
```
Comment (jsjoeio) [inline jest.config.js:5]: 🔨 Basic Jest config (as basic as it gets)

**Comment by kennethcassel** [comment]
Thank you @jsjoeio and twitch streaming crew :)

**Comment by kennethcassel** [comment]
@all-contributors can you please add @jsjoeio for code contributions :)

**Pull request merged**